src/basic/log.h

```diff
@@ -44,14 +54,24 @@ typedef enum LogTarget{
         LOG_TARGET_NULL,
         _LOG_TARGET_MAX,
         _LOG_TARGET_INVALID = -1
-}  LogTarget;
+} LogTarget;
+
+#define REALM_PLUS_LEVEL(realm, level)          \
+        ((realm) << 10 | (level))
+#define REALM(realm_level)                      \
+        ((realm_level >> 10))
```

Comment:
hmm, could you prefix these two macros with `LOG_` please? all the other stuff we export from this header has such a prefix